### PR TITLE
add possibility to upload binary file via uri module - Ansible 1.9 backport

### PR DESCRIPTION
**Pull request by mploski**
##### ISSUE TYPE

Feature Pull Request
##### COMPONENT NAME

URI
##### ANSIBLE VERSION

1.9
##### SUMMARY

This feature allow to send binary files to endpoint smoothly. This is backport to Ansible < 2.0
Added new parameter source.
When source parameter is valid file is open and file descriptor is passed to URLLIB2.

https://github.com/ansible/ansible-modules-core/issues/3157

```
```

### CODE REVIEW

File: network/basics/uri.py [hunk 1]
```diff
@@ -37,7 +36,7 @@
 description:
   - Interacts with HTTP and HTTPS web services and supports Digest, Basic and WSSE
     HTTP authentication mechanisms.
-version_added: "1.1"
+version_added: "1.9"
```
Comment (sivel) [inline network/basics/uri.py]: This `version_added` should not be changed, it was the version that the `uri` module was first added in

File: network/basics/uri.py [hunk 2]
```diff
@@ -49,6 +48,12 @@
       - path of where to download the file to (if desired). If I(dest) is a directory, the basename of the file on the remote server will be used.
     required: false
     default: null
+  source:
+    description:
+      - path to binary file to be uploaded to the endpoint
+    required: false
+    default: null
+    version_added: '1.9.7'
```
Comment (sivel) [inline network/basics/uri.py:56]: This version should be the next major release version, which at this point should be likely `2.3`

File: network/basics/uri.py (hunk 1 above)
Comment (mploski) [reply network/basics/uri.py]: Reverted

File: network/basics/uri.py (hunk 2 above)
Comment (sivel) [reply network/basics/uri.py:56]: Oops, misunderstood that this was a backport.  This is probably fine.

Comment (mploski) [reply network/basics/uri.py:56]: So is there no possibility to backport this changes to ansible < 2.0 merge it to upstream and release it as a minor release ?

**Comment by gregdek** [comment]
Thanks @mludy. All backport requests must be reviewed by the core team, and this can take time. We appreciate your patience. 

[This message brought to you by your friendly Ansibull-bot.]

**Comment by bcoca** [comment]
We are not planing making any new 1.9 releases, even if we did only bugfixes would go in. 

Any new features should be geared to newer releases.

**Pull request closed without merging**